nevergrad/optimization/optimizerlib.py

```diff
@@ -1082,14 +1082,14 @@ def learn_on_k_best(archive: utils.Archive[utils.MultiValue], k: int) -> tp.Arra
     model = LinearRegression()
     model.fit(X2, y)
 
-    # Find the minimum of the quadratic model.
-    optimizer = OnePlusOne(parametrization=dimension, budget=dimension * dimension + dimension + 500)
+    optimizer = Powell(parametrization=dimension, budget=45*dimension+30)
     try:
-        optimizer.minimize(lambda x: float(model.predict(polynomial_features.fit_transform(np.asarray([x])))))
+        target = lambda x: float(model.predict(polynomial_features.fit_transform(np.asarray([x]))))
```

Comment:
move this out of the try, the try expect close should be as small as possible
also, why is is converted to array since it already is one? to reshape it ? then just do x[None, :] or x[:, None] depending on the required shape.
Affecting a lambda to a variable is usually not a good idea either, but nevermind for now

Reply:
Affectation of the lambda removed.
Conversion to array removed.
Moved out of the try.